src/mainloop.cpp

```diff
@@ -217,14 +215,27 @@ void Mainloop::camera_callback(const void *img, size_t len, const struct timeval
 
 	Point3_<double> gyro_data;
 	struct timespec gyro_timespec;
-	_bmi->gyro_integrated_get(&gyro_data, &gyro_timespec);
+	static struct timespec _gyro_last_timespec = gyro_timespec;
+	static Point3_<double> gyro_integrated_data(0.0, 0.0, 0.0);
+
+	_bmi->get_gyro_sample(&gyro_data, &gyro_timespec);
```

Comment:
It's not that important but yes you're right, it would be cleaner if it integrates all of them and then a reset function gets called once the flow is published